pubsub/gochannel/pubsub.go

```diff
@@ -370,6 +370,11 @@ SendToSubscriber:
 			return
 		}
 
+		if !blockPublishUntilSubscriberAck {
+			s.logger.Trace("Sent message to subscriber without ack", logFields)
+			return
+		}
+
```

Comment:
The select below isn't really blocking execution, as it's running in another goroutine. `BlockPublishUntilSubscriberAck` is checked in `Publish` and that's where it blocks.

Can you share more about your use case?

Reply:
@m110 the current implementation will cause deadlock on all subscribers even if one of them did Ack/Nack a message. However I believe only the subscriber that does not Ack/Nack should lose the message leaving the rest processing goes through. So the setting to block publishing will deadlock if one subscriber does not Ack and I don't think that the intention.

Reply:
My suggestion will be to timeout when the `BlockPublishUntilSubscriberAck` is set to avoid the deadlock